Doc/Manual/Fortran.html

```diff
@@ -0,0 +1,2275 @@
+
+<p>
+Some functions may include references or pointers to classes that are not wrapped by Fortran proxy functions. In these cases, an opaque derived type called <code>SwigUnknownClass</code> will be generated and used as a placeholder for the argument or return value. These could theoretically be passed between wrapped SWIG functions, although no type checking will be performed to ensure that the unknown classes are the correct types.
+</p>
+
+
+<p>
+This behavior will change soon to enforce type safety.
+</p>
+
+<H3><a name="Fortran_enumerations">Enumerations</a></H3>
+
+<p>
+Fortran 2003 implements C enumerations using the <code>ENUM, BIND(C)</code> statement. These enumerators are simply a set of loosely grouped compile-time integer constants that are guaranteed to be compatible with C enumerators. Unlike C++, all enumerators in Fortran are anonymous.
+</p>
+
+
+<p>
+To associate a C enumeration name with the Fortran generated wrappers, SWIG generates an additional enumeration with the C class name and a dummy value of <code>-1</code>. The enumeration generated from the C code
+</p>
+
+
+<div class="code"><pre><code>enum MyEnum {
+  RED = 0,
+  GREEN,
+  BLUE,
+  BLACK = -1
+};</code></pre></div>
+
+
+<p>
+looks like:
+</p>
+
+
+<div class="targetlang"><pre><code> enum, bind(c)
+  enumerator :: MyEnum = -1
+  enumerator :: RED = 0
+  enumerator :: GREEN = RED + 1
+  enumerator :: BLUE = GREEN + 1
+  enumerator :: BLACK = -1
+ end enum</code></pre></div>
+
+
+<p>
+These enumerators are treated as standard C integers in the C wrapper code code. In the Fortran wrapper code, procedures that use the enumeration use the type <code>integer(kind(MyEnum))</code> to clearly indicate what enum type is required.
+</p>
+
+
+<p>
+Some C++ enumeration definitions cannot be natively interpreted by a Fortran compiler (e.g. <code>FOO = 0x12,</code> or <code>BAR = sizeof(int),</code>), so these are defined in the C++ wrapper code and <em>bound</em> in the Fortran wrapper code:
+</p>
+
+
+<div class="targetlang"><pre><code>integer(C_INT), protected, public, &amp;
+   bind(C, name=&quot;swigc_FOO&quot;) :: FOO</code></pre></div>
+
+
+<p>
+The <code>%fortranconst</code> can be used to explicitly enable and disable treatment of a C++ <code>enum</code> as a Fortran enumerator; the <code>%nofortranconst</code> directive forces the values to be wrapped as externally-bound C integers.
+</p>
+
+
+<p>
+If an enumeration type has not been defined but is used in a function signature, a placeholder <code>SwigUnknownEnum</code> enumerator will be generated and used instead.
+</p>
+
+
+<p>
+Class-scoped enumerations are prefixed with the class name:
+</p>
+
+
+<div class="code"><pre><code>struct MyStruct {
+  enum Foo {
+    Bar = 0
+  };
+};</code></pre></div>
+
+
+<p>
+generates
+</p>
+
+
+<div class="targetlang"><pre><code>enum, bind(c)
+ enumerator :: MyStruct_Foo = -1
+ enumerator :: MyStruct_Bar = 0
```

Comment:
Good to see enhanced enum documentation. I was wondering why there is a `MyStruct_Foo = -1` ? This is not an enum value in C and from what I can make out in Fortran enums, isn't necessary. It isn't something that SWIG normally generates either.

Reply:
Also, I can't get %fortranconst and %nofortranconst to make any difference for enums. The docs really need more examples like for this feature. If you put one in, I might be able to work out what I'm doing wrong, but I think it may be buggy.